### PR TITLE
test: add tests for parsers._moduleNameFromFrames

### PR DESCRIPTION
**Pull request by trentm**
Also refactor lib/parsers.js slightly. No functional change.

Fixes: #1918

### CODE REVIEW

**Comment by trentm** [comment]
Coverage before:

```
----------------------------------------------|---------|----------|---------|---------|-------------------
File                                          | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
----------------------------------------------|---------|----------|---------|---------|-------------------
All files                                     |   84.62 |     73.7 |   89.31 |   85.95 |
...
 apm-agent-nodejs/lib                         |   91.23 |    84.33 |   99.07 |   92.18 |
...
  parsers.js                                  |    88.3 |    78.99 |     100 |    89.1 | 48,89,201-223,293
```

After:

```
$ npm run coverage
---------------------------------|---------|----------|---------|---------|-------------------
File                             | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
---------------------------------|---------|----------|---------|---------|-------------------
All files                        |   84.59 |    73.66 |   89.31 |   85.92 |
...
 apm-agent-nodejs3/lib           |   91.15 |    84.12 |   99.07 |    92.1 |
...
  parsers.js                     |    87.8 |    78.26 |     100 |   88.59 | 48,89,201-223,295
...
```

**Comment by trentm** [comment]
One incomplete check "Integration Tests Pending — Integration Tests ... "  after many hours. I don't know why.

**Comment by trentm** [comment]
jenkins run the tests

File: lib/parsers.js [hunk 1]
```diff
@@ -14,11 +14,11 @@ var stacktrace = require('error-stack-parser')
```
Comment (astorm) [inline lib/parsers.js:14]: I see we've modified this module to use the "export everything at the end" style, changed function definitions to be in the `function functionName` style, and changed the formally local-to-this-module `getModule` to `_moduleNameFromFrames` and are now exporting this function in order to test it.

**Review by astorm — APPROVED** [review]
It does the thing and tests cover the common use cases.  Approving.

File: lib/parsers.js (hunk 1 above)
Comment (trentm) [reply lib/parsers.js:14]: @astorm I hope you are good with those changes. 

- I noticed you and I both perhaps gravitated to that style for module.exports (from recently added lib/logging.js and lib/wildcard-matcher.js).
- The `fooFromBar` is naming style if grown fond of. The influence for me was https://www.joelonsoftware.com/2005/05/11/making-wrong-code-look-wrong/

**Pull request merged**